core/src/main/java/io/ddf/DDFCache.java

```diff
@@ -2,52 +2,90 @@
 
 
 import com.google.common.base.Strings;
+import com.google.common.cache.*;
 import io.ddf.exception.DDFException;
-
 import java.util.Map;
 import java.util.UUID;
 import java.util.concurrent.ConcurrentHashMap;
 
+import io.ddf.misc.ALoggable;
+import io.ddf.misc.Config;
+import org.apache.commons.lang.exception.ExceptionUtils;
+
+import java.util.concurrent.TimeUnit;
+
 /**
- * Created by huandao on 6/11/15.
  */
-public class DDFCache {
+public class DDFCache extends ALoggable {
+
+  private LoadingCache<UUID, DDF> mDDFCache;
+
+  private DDFManager mDDFManager;
+
+  public DDFCache(DDFManager manager) {
+    mDDFManager = manager;
+    /**
+     * Maximum number of DDFs allowed in cache, DDFs will be evicted by an approximate LRU algorithm
+     */
+    Long maxNumberOfDDFs = Long.valueOf(Config.getGlobalValue(Config.ConfigConstant.MAX_NUMBER_OF_DDFS_IN_CACHE));
+
+    /**
+     * The number of seconds that the DDF will live in cache since it last used(meaning inserted or accessed)
+     */
+    Long timeToIdle = Long.valueOf(Config.getGlobalValue(Config.ConfigConstant.DDF_TIME_TO_IDLE_SECONDS));
+
+    mLog.info(String.format("Maximum number of ddfs in cache %s", maxNumberOfDDFs));
+    mLog.info(String.format("DDF's time to idle = %s seconds", timeToIdle));
+    mDDFCache = CacheBuilder.newBuilder().
+        maximumSize(maxNumberOfDDFs).expireAfterAccess(timeToIdle, TimeUnit.SECONDS).
```

Comment:
@Huandao0812: MaximumSize usually use for size-base eviction, expireAfterAccess other hand use with time-base eviction, I don't think it's good to use both. 
Only maximumSize is enough, it will use LRU to evict by default.